### DISCUSSION TITLE
www.dobreprogramy.pl

### DISCUSSION THREAD

**Discussion by admin** [FiltersHeroes/PolishCookieConsent#124] [Pytania i odp] [answered]
W Firefox funkcja klikania w "przechodz" nie chce działać z mechanizmem ochrony prywatności.

> ![obraz](https://user-images.githubusercontent.com/35370833/60910807-13042e00-a282-11e9-87ae-ed458e7a0451.png) |
> --- |

Pewnie trzeba by po prostu innym dodatkiem blokować ciasteczka a nie mechanizmem Firefox?

**Comment by admin** [comment, marked as answer]
@krystian3w Nie chce działać, bo pewnie ustawienia są zapisywane do :cookie:, więc musiałbyś zrezygnować z blokowania :cookie: na tej stronie. Także to nie jest błąd, tak ma być.

**Comment by seba2282** [comment]
Odnośnie tego, Do Not Track, ochrona śledzenia tylko w kartach prywatnych i akceptacja wszystkich cookies zezwolone i na podobnych ustawieniach działa z chromowym Edge, a z Firefox już nie.

**Comment by admin** [comment]
Jeszcze jest opcja że masz za dużo dodatków lub twoje milion list tak zamula przeglądarkę że nawet 1000 ms opóźnienia to za mało.

**Comment by admin** [comment]
W sumie i tak najlepsze jest, że WP śledzi klikania Enterów i po tym niby rozpoznaje użytkownika?

https://forum.dobreprogramy.pl/t/portal-komentarze-enter-2-bledy/568397/27

**Comment by seba2282** [comment]
Ublock origin z tymi samymi filtrami w obu przeglądarkach + PZC

**Comment by admin** [comment]
Jeżeli opcja dot. :cookie: jest odznaczona, to powinno być ok. No chyba, że @seba2282 wybrał, że ma wszystkie :cookie: blokować albo dopisał coś w wyjątkach w Ciasteczka i dane stron, że ma coś blokować :smiley:

**Comment by seba2282** [comment]
Napisałem, mam zezwól na wszystkie cookies.

**Comment by admin** [comment]
Znowu się zepsuło, wrócili do zielonego paska z białego wyśrodkowanego pop-upa, pewnie jakiś bug że aktywował się biały.

Button niby wygląda OK:

```
<button class="n4k1a2v" n4k1a2v="">PRZECHODZĘ DO&nbsp;SERWISU</button>
```

Skrypt **kiboke** dalej umie kliknąć: https://github.com/PolishFiltersTeam/PolishCookieConsent/issues/19


```js
function _parent(e)
{
	if (e && e.parentNode)
		return e.parentNode;
	
	return false;
}

function _id(s)
{
	return document.getElementById(s);
}

function _sl(s, c)
{
	return (c || document).querySelector(s);
}

function _ev(s, c)
{
	return document.evaluate("//"+s, c || document, null, XPathResult.ANY_TYPE, null).iterateNext();
}


function getE(h)
{
	switch (h)
	{
		case 'dobreprogramy.pl':
			return _ev("button[contains(., 'PRZECHODZ')]");
	}
	
	
	var parts = h.split('.');
	
	if (parts.length > 2)
	{
		parts.shift();
		return getE(parts.join('.'));
	}
	
	return false;
}


// Search loop function

function searchLoop(counter, host) {
	setTimeout(function() {
		var e = getE(host);
		
		if (e)
			e.click();
		else if (counter < 200)
			searchLoop(counter+1, host);
	}, 500);
}


// Initial timeout

setTimeout(function(){
	searchLoop(0, document.location.hostname.replace(/^w{2,3}\d*\./i, ''));
}, 500);
```

niby zwraca dwucyfrową liczbę np. `89`.

**Comment by admin** [comment]
Konsola www mówi że element nie jest zdefiniowany - lub że z konsoli nigdy nie zadziała (nie kliknę nawet w coś co widać):

![obraz](https://user-images.githubusercontent.com/35370833/62836483-bc648800-bc63-11e9-925f-10c76f945bb4.png) |
--- |

**Comment by seba2282** [comment]
No popatrz, nie tylko mnie denerwuje RODO/cookie na DP 😂 A tak poważnie, kiboke ma trzy opcje w I don't care about cookies (zezwalaj na wszystkie, zezwalaj na cookies potrzebne do poprawnego działania strony, blokuj wszystkie cookies). Natomiast lista sobie nadal nie radzi.

**Comment by admin** [comment]
> Konsola www mówi że element nie jest zdefiniowany - lub że z konsoli nigdy nie zadziała (nie kliknę nawet w coś co widać):

Najwyraźniej pomiędzy button musi być cudzysłów, bo inaczej nie przechodzi, ale pewnie to nie błąd FF, tylko tak miało być, a Chromium/Vivaldi nie sprawdza dokładnie.

**Comment by admin** [comment]
@krystian3w Testuj, debuguj, czy teraz lepiej. 
Jednak chyba nie chodziło o cudzysłów, tylko na FF coś nie chciała działać ta opcja z timeout, więc zrobiłem inaczej.

`Natomiast lista sobie nadal nie radzi.`
I nie będzie, bo klikanie działa tylko w Nano i AG dla uprzywilejowanych list.

**Comment by seba2282** [comment]
W AG już nie działa dla głównej DP, nano też nie ogarnia. I pies pogrzebany.

**Comment by admin** [comment]
@seba2282 Przetestuj z ostatniego commita. Pobierz repo, wejdź na `about:debugging#addons` i tam `tymczasowo wczytaj` i wskaż manifest.

**Comment by admin** [comment]
W AG jak ma ruszyć to tylko z AG 3.2.2 lub nowszym (FX / Opera) nie wiem jak Edge podchodzi do skryptów dynamicznie pobieranych  - może olewają jak Chrome.

**Comment by seba2282** [comment]
AG stable 3.2.1 od pewnego czasu nie blokuje. Dlatego przetestuję za moment wyższe wersje. Jak tu będzie problem na FF, to nie wiem co dalej.

**Comment by admin** [comment]
@krystian3w @seba2282 Nadal czekam na wasze wyniki :smile:.

**Comment by admin** [comment]
Jak dla mnie poprawka może być, sprawdziłem w ViolentMonkey i kliknął. A dodatek minimalnie szybciej.

**Comment by admin** [comment]
Już jest w AMO nowa wersja.

**Comment by seba2282** [comment]
Sorki, że tak późno, poprawka zadziałała. Tylko do potwierdzenia.

**Comment by admin** [comment]
`Chro-Edge` może do czwartku dostanie. Tak by kody były identyczne a nie że zbugowany tam się odpalał.

**Comment by admin** [comment]
@krystian3w A, czyli wyłożyłeś kasę dla MS i będzie w sklepie MS?

**Comment by admin** [comment]
Nie trzeba - niebieski pasek że pozwalasz na z chrome web store, raczej jeszcze tego MS nie zablokował.

[a comment by an automated account is not shown]
